### DISCUSSION TITLE
Downloads of subs only happens/shows up after docker restart.

### DISCUSSION THREAD

**Discussion by rustjakey** [Tzahi12345/YoutubeDL-Material#337] [Q&A] [answered]
When I add a channel for example, nothing will happen, no downloads will start, or at least appear to. 

I then restart the docker about 10 mins later and a few will show up.

Any ideas about this?

Thanks

**Comment by Tzahi12345** [comment]
Here's the TLDR: we only know which videos were downloaded after the entire channel is checked, meaning you have to wait until it's all done for it to be registered into the UI.

When you restart the container, all files are checked to see if any of them don't exist in the DB. If they are missing they are added. So this is the behavior you are seeing.

One alternative we can do is if a channel is being downloaded/checked, we can run that logic in that folder to auto add files as they are being downloaded. Shouldn't take too long to implement, let me know if this is something that would be helpful in your use case or if waiting isn't an issue (for large channels it absolutely makes sense).

**Reply by rustjakey** [reply]
Thank you for the reply and for the work you have done. 

I do not see why not to add the check, it would only be helpful, unless there are issues you haven't mentioned. 

On a side note, I have a really hard time getting playlists to work, they will only ever download the first song. I've tried all sorts of ways of getting a links. Is there anything you suggest, or it needs to work?  

Thanks

**Reply by Tzahi12345** [reply]
I'll work on getting that workaround added. For the playlists issue, which version of YoutubeDL-Material are you running? Do you mind giving the playlist URL so I can test it on my end? On the latest nightly it works as expected.

**Comment by rustjakey** [comment]
This is a link I got off my phone, so it might work as I haven't tried I
before

https://youtube.com/playlist?list=PLfvpqdsXvlf4vqsRWglvSPplpKkEsBI7E

I am on the latest version available, from the docker page as far as I know.

**Comment by rustjakey** [comment]
@Tzahi12345 Any ETA on this implementation. I am now finding that it will never update to show downloads uless i restart it. 

Thanks

**Comment by Tzahi12345** [comment]
Hi, sorry for the delayed response, been very busy this last week. Would you mind creating an issue as a reminder? Hopefully I can get to it this week

**Comment by Tzahi12345** [comment, marked as answer]
Just wanted to drop in and say this has been fixed for a couple months now! As a subscription downloads videos they now get automatically added as they are detected.
